lib/src/rules/unnecessary_const.dart

```diff
@@ -60,13 +62,9 @@ class _Visitor extends SimpleAstVisitor {
   void visitInstanceCreationExpression(InstanceCreationExpression node) {
     if (node.keyword == null) return;
 
-    // remove keyword and check if there's const error
-    final oldKeyword = node.keyword;
-    node.keyword = null;
-    final isConstWithoutKeyword = node.isConst;
-    node.keyword = oldKeyword;
-
-    if (isConstWithoutKeyword && node.keyword?.type == Keyword.CONST) {
+    // todo (pq): remove cast when `inConstantContext` is public on Element
+    var inConstContext = (node as ExpressionImpl).inConstantContext;
+    if (inConstContext && node.keyword?.type == Keyword.CONST) {
```

Comment:
I would have switched the order, and checked `node.keyword?.type == Keyword.CONST` first (as you have in the first statement of this method). It is cheaper.